app/fields/date_field.rb

```diff
@@ -35,4 +35,14 @@ def self.from_persisted_value(date)
     end
   end
 
+  def set_error_message_lookup_proc!(error_proc)
+    @error_message_lookup_proc = error_proc
+  end
+
+  private
+
+  def error_message_lookup_proc
+    @error_message_lookup_proc || ->(a){ a }
```

Comment:
It will work, I just wanted to check what the intention is. I was just thinking that if this method gets called 20 times for the form, we could cache it. But mostly I guess it'll be set explicitly by the setter..